Add Tabs .getCurrent and .onTabSelectionChanged methods for Chrome and FF

ssawchenko opened the pull request:
I have added in the tabs getCurrent and onTabSelectionChanged methods for Chrome and FF. 
onTabSelectionChanged does not appear to work correctly in FF; success callback only appears to be correctly fired once, despite the fact that the activate event is being correctly fired for each tab change.

Also note, I have reverted the change for the first commit; I realized that my local history still contained this commit.

ssawchenko committed:
Add support for local button icons

* Use data.url to patch up local icon path

ssawchenko committed:
Added tabs getCurrent and onTabSelectionChanged

* Note, getCurrent appears to work in Chrome and FF; IE
  implementation still required
* onTabSelection appears to work in Chrome, but not FF; IE
  implementation still required
* Reverted last change for setIcon

reicolina commented:
Good work! but probably best not to merge to master until the FF issue is fixed and the IE functionality for those methods gets implemented. Also, can you rename this pull request with a more descriptive subject? That way, it is clear what we want to achieve/fix once we do a pull request to the original trigger.io repo.

ssawchenko commented:
Sounds good - I will wait for the master check in until the functionality is fully complete. This check in was more to make the code available for people to review in hopes that a fresh set of eyes can help solve the FF issue I am seeing. Thanks Rei :)

ssawchenko committed:
Further fixes for onTabSelectionChanged

* onTabSelection appears to work in Chrome, but FF requires that
  the method be called each time the user wishes to listen for the next
  tab change; IE implementation still required

ssawchenko committed:
Added tabs.getAllTabs for Chrome and FF
